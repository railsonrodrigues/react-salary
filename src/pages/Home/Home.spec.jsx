import React from 'react';
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event';
import { render, screen, waitFor } from "@testing-library/react";
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import HomePage from './Home';

describe('<HomePage />', () => {
  it('should render HomePage correctly', () => {
    const { container, debug } = render(
      // <Router>
      //   <Switch>
      //     <Route path="/home">
                <HomePage />
      //     </Route>
      //   </Switch>
      // </Router>
    );
    const title = screen.getByRole('heading', {name: 'Calcule Seu Salário Líquido'})
    expect(title).toBeInTheDocument();

    const form = screen.getByRole('form');
    expect(form).toBeInTheDocument();

    const cards = container.querySelectorAll('.eHEsiy');
    expect(cards).toHaveLength(2);

    const flatCards = container.querySelectorAll('.fkoXlF');
    expect(flatCards).toHaveLength(3);

    const graphic = container.querySelector('.graphic');
    expect(graphic).toBeInTheDocument();
  })
  it('should match snapshot', () => {
    const { container } = render(
      // <Router>
      //   <Switch>
      //     <Route path="/home">
                <HomePage />
      //     </Route>
      //   </Switch>
      // </Router>
    );
    expect(container).toMatchSnapshot();
  })
  it('should change cards values on submit', async () => {
    const { container, debug } = render(<HomePage />);
    const inputGrossSalary = screen.getByPlaceholderText('Salário bruto')
    const inputnDependents = screen.getByPlaceholderText('Número de dependentes')
    const inputOtherDiscounts = screen.getByPlaceholderText('Outros Descontos')
    const buttonSubmit = screen.getByRole('button', {name: /calcular/i})

    userEvent.type(inputGrossSalary, '4500');
    userEvent.type(inputnDependents, '0');
    userEvent.type(inputOtherDiscounts, '0');
    userEvent.click(buttonSubmit);
    debug();
    const cardNetSalary = container.querySelector('.bslhuf');
    const value = 'R$ 0,00'
    console.log(value === 'R$ 0,00')
    expect(cardNetSalary.textContent).toBe('R$ 3.750,64')
  })
})