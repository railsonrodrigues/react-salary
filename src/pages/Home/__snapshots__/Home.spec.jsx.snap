// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should match snapshot 1`] = `
<div>
  <h1
    class="sc-crzoAE dKaBpI"
  >
    Calcule Seu Salário Líquido
  </h1>
  <div
    class="sc-bdnxRM iewkLq"
  >
    <form
      role="form"
    >
      <div
        class="sc-kEqXSa kaeJao"
      >
        <input
          class="sc-pNWdM kIPrmQ"
          id="grossSalary"
          placeholder="Salário bruto"
          type="text"
        />
        <input
          class="sc-pNWdM kIPrmQ"
          id="nDependents"
          placeholder="Número de dependentes"
          type="text"
        />
        <input
          class="sc-pNWdM kIPrmQ"
          id="otherDiscounts"
          placeholder="Outros Descontos"
          type="text"
        />
      </div>
      <div
        class="sc-iqAclL fJCKud"
      >
        <button
          class="sc-jrsJWt jazCUG"
        >
          Calcular
        </button>
      </div>
    </form>
  </div>
  <div
    class="sc-bdnxRM iewkLq"
  >
    <div
      class="sc-gtsrHT iGvWbg"
    >
      <div
        class="sc-dlnjwi eHEsiy"
      >
        <h3
          class="sc-hKFxyN gYuxsg"
        >
          Salário líquido
        </h3>
        <span
          class="sc-eCApnc bslhuf"
        >
          R$ 0,00
        </span>
        <span
          class="sc-jSFjdj binByy"
        >
          0.0%
        </span>
      </div>
      <div
        class="sc-dlnjwi eHEsiy"
      >
        <h3
          class="sc-hKFxyN gYuxsg"
        >
          Desconto total
        </h3>
        <span
          class="sc-eCApnc dHwNeS"
        >
          R$ 0,00
        </span>
        <span
          class="sc-jSFjdj binByy"
        >
          0.0%
        </span>
      </div>
    </div>
  </div>
  <div
    class="sc-bdnxRM iewkLq"
  >
    <div
      class="sc-gtsrHT GcGKx"
    >
      <div
        class="sc-gKAaRy fkoXlF"
      >
        <h3
          class="sc-fujyAs cYyfKX"
        >
          INSS
        </h3>
        <span
          class="sc-iCoGMd hCYTca"
        >
          R$ 0,00
        </span>
        <span
          class="sc-iCoGMd cuEIYz"
        >
          0.0%
        </span>
      </div>
      <div
        class="sc-gKAaRy fkoXlF"
      >
        <h3
          class="sc-fujyAs cYyfKX"
        >
          IRRF
        </h3>
        <span
          class="sc-iCoGMd hCYTca"
        >
          R$ 0,00
        </span>
        <span
          class="sc-iCoGMd cuEIYz"
        >
          0.0%
        </span>
      </div>
      <div
        class="sc-gKAaRy fkoXlF"
      >
        <h3
          class="sc-fujyAs cYyfKX"
        >
          Outros
        </h3>
        <span
          class="sc-iCoGMd hCYTca"
        >
          R$ 0,00
        </span>
        <span
          class="sc-iCoGMd cuEIYz"
        >
          0.0%
        </span>
      </div>
    </div>
  </div>
  <div
    class="sc-bdnxRM iewkLq"
  >
    <div
      class="graphic"
    />
  </div>
</div>
`;
